caffe2/onnx/onnx_exporter.h

```diff
@@ -81,6 +90,9 @@ class OnnxExporter {
       get_per_op_renamed_attrs() const;
   const std::unordered_map<std::string, OnnxExporter::SpecialOpConverter>&
   get_special_operators() const;
+
+  // Dummy name generator
+  std::shared_ptr<DummyName> dummy_;
```

Comment:
I think there is no need to use shared_ptr here, maybe just a simple instance?

Reply:
There is a problem. This class can have its own `DummyName` or it can get it from Python interface. When it's from Python interface we don't claim ownership. Therefore I use `shared_ptr` with customized dtor to handle both cases.

Reply:
Ok I think this can be cleaned up once we have the registration style converters?

Reply:
If we get rid of the python dependency, then yes.